src/main.rs

```diff
@@ -1,3 +1,132 @@
+#[macro_use]
+extern crate clap;
+
+mod options;
+use crate::options::CLIOptions;
+use clap::App;
+use std::{fs, fs::File, io, path::Path};
+
 fn main() {
-    println!("Hello, world!!!");
+    let yaml = load_yaml!("../cli_def/en_us.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    let raw_options = CLIOptions::from_clap_matches(&matches);
+
+    let options = raw_options.process_options();
+
+    for filename in options.files {
+        let file = File::open(filename).unwrap();
+        unzip_file(
+            file,
+            &std::path::PathBuf::from(&raw_options.destination_folder),
+        );
+    }
+}
+
+fn unzip_file(file: File, directory: &Path) {
+    let mut archive = zip::ZipArchive::new(file).unwrap();
+
+    let files_indexes = 0..archive.len();
+
+    files_indexes
+        .map(|file_index| unzip_archive(file_index, &mut archive, directory))
+        .for_each(drop);
+}
+
+fn unzip_archive(file_index: usize, archive: &mut zip::ZipArchive<File>, directory: &Path) {
+    let mut file = archive.by_index(file_index).unwrap();
+    let outpath = file.sanitized_name();
+
+    {
+        let comment = file.comment();
+        if !comment.is_empty() {
+            println!("File {} comment: {}", file_index, comment);
+        }
+    }
+
+    if (&*file.name()).ends_with('/') {
+        println!(
+            "File {} extracted to \"{}\"",
+            file_index,
+            outpath.as_path().display()
+        );
+        fs::create_dir_all(directory.join(&outpath)).unwrap();
+    } else {
+        println!(
+            "File {} extracted to \"{}\" ({} bytes)",
+            file_index,
+            outpath.as_path().display(),
+            file.size()
+        );
+        if let Some(p) = outpath.parent() {
```

Comment:
a match can be a bit cleaner this way as you can have the condition of `!p.exists()` in there. It will make it easier to read.